tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CustomizedCodeUpdateTool/TspClientUpdateTool.cs

```diff
@@ -0,0 +1,387 @@
+    {
+        // The sub-command handlers already perform logic.
+        await Task.CompletedTask;
+    }
+
+    // --------------- Response Models ---------------
+    public class UpdateSessionState
+    {
+        [JsonPropertyName("sessionId")] public string SessionId { get; set; } = Guid.NewGuid().ToString("n");
+        [JsonPropertyName("specPath")] public string SpecPath { get; set; } = string.Empty;
+        [JsonPropertyName("oldGeneratedPath")] public string OldGeneratedPath { get; set; } = string.Empty;
+        [JsonPropertyName("newGeneratedPath")] public string NewGeneratedPath { get; set; } = string.Empty;
+        [JsonPropertyName("status")] public string Status { get; set; } = "Initialized";
+        [JsonPropertyName("apiChanges")] public List<ApiChange> ApiChanges { get; set; } = new();
+        [JsonPropertyName("impactedCustomizations")] public List<CustomizationImpact> ImpactedCustomizations { get; set; } = new();
+        [JsonPropertyName("directMergeFiles")] public List<string> DirectMergeFiles { get; set; } = new();
+        [JsonPropertyName("proposedPatches")] public List<PatchProposal> ProposedPatches { get; set; } = new();
+    }
+
+    public class ApiChange
+    {
+        [JsonPropertyName("kind")] public string Kind { get; set; } = string.Empty;
+        [JsonPropertyName("symbol")] public string Symbol { get; set; } = string.Empty;
+        [JsonPropertyName("detail")] public string Detail { get; set; } = string.Empty;
+    }
+
+    public class CustomizationImpact
+    {
+        [JsonPropertyName("file")] public string File { get; set; } = string.Empty;
+        [JsonPropertyName("reasons")] public List<string> Reasons { get; set; } = new();
+    }
+
+    public class PatchProposal
+    {
+        [JsonPropertyName("file")] public string File { get; set; } = string.Empty;
+        [JsonPropertyName("diff")] public string Diff { get; set; } = string.Empty; // unified diff placeholder
+        [JsonPropertyName("rationale")] public string Rationale { get; set; } = string.Empty;
+    }
+
+    // Generic response wrapper (could reuse DefaultCommandResponse but keeping custom for clarity)
+    public class UpdateResponse : Response
+    {
+        [JsonPropertyName("session")] public UpdateSessionState? Session { get; set; }
+        [JsonPropertyName("message")] public string? Message { get; set; }
+        public override string ToString()
+        {
+            var sb = new System.Text.StringBuilder();
+            if (!string.IsNullOrEmpty(Message)) sb.AppendLine(Message);
+            if (Session != null) sb.AppendLine($"Session: {Session.SessionId} Status: {Session.Status} API changes: {Session.ApiChanges.Count}");
+            return ToString(sb);
+        }
+    }
+
+    // --------------- MCP Methods ---------------
+
+    [McpServerTool(Name = "tsp_update_regenerate"), Description("Stage1 regenerate placeholder")]
```

Comment:
I don't think this name and description are conveying it's use easily. And name should also starts with azsdk_